lib/ansible/modules/windows/win_wait_for_process.ps1

```diff
@@ -0,0 +1,152 @@
+{
+    Fail-json $result "process_min_count parameter is of no effect when waiting for a process to stop."
+}
+
+if (($process_name_exact -or $process_name_pattern) -and $process_id)
+{
+    Fail-json $result "process_id may not be used with process_name_exact or process_name_pattern"
+}
+if ($process_name_exact -and $process_name_pattern)
+{
+    Fail-json $result "process_name_exact and process_name_pattern may not be used at the same time."
+}
+
+if (-not ($process_name_exact -or $process_name_pattern -or $process_id -or $owner))
+{
+    Fail-json $result "at least one of: process_name_exact, process_name_pattern, process_id, or user must be supplied"
+}
+
+$module_start = Get-Date
+
+#Get-Process doesn't actually return a UserName value, so get it from WMI.
+Function Get-ProcessMatchesFilter {
+    [cmdletbinding()]
+    Param(
+        [String]
+        $Owner,
+        $ProcessNameExact,
+        $ProcessNamePattern,
+        [int]
+        $ProcessId
+    )
+    
+    $CIMProcesses = Get-CimInstance Win32_Process
+    foreach ($CIMProcess in $CIMProcesses)
+    {
+        $include = $true
+        if(-not [String]::IsNullOrEmpty($ProcessNamePattern))
+        {
+            #if a process name was specified in the filter, validate that here.
+            $include = $include -and ($CIMProcess.ProcessName -match $ProcessNamePattern)
+        }
+        if($ProcessNameExact -is [Array] -or (-not [String]::IsNullOrEmpty($ProcessNameExact)))
+        {
+            #if a process name was specified in the filter, validate that here.
+            if ($ProcessNameExact -is [Array] )
+            {
+                $include = $include -and ($ProcessNameExact -contains $CIMProcess.ProcessName)
+            }   
+            else {
+                $include = $include -and ($ProcessNameExact -eq $CIMProcess.ProcessName)
+            }
+        }
+        if ($ProcessId -and $ProcessId -ne 0)
+        {
+            # if a PID was specified in the filger, validate that here.
+            $include = $include -and ($CIMProcess.ProcessId -eq $ProcessId)
+        }
+        if (-not [String]::IsNullOrEmpty($Owner) )
+        {
+            # if an owner was specified in the filter, validate that here.
+            $include = $include -and ($($(Invoke-CimMethod -InputObject $CIMProcess -MethodName GetOwner).User) -eq $Owner)
+        }
+        
+        if ($include)
+        {
+            $CIMProcess | Select-Object -Property ProcessId, ProcessName, @{name="Owner";Expression={$($(Invoke-CimMethod -InputObject $CIMProcess -MethodName GetOwner).User)}}
+        }
+    }
+}
+
+Start-Sleep -Seconds $pre_wait_delay
+if ($state -eq "present" ) {
+    #wait for a process to start
+    $Processes = @()
+    $attempts = 0
+    Do {
+        if (((Get-Date) - $module_start).TotalSeconds -gt $timeout)
+        {
+            Fail-Json $result "timeout while waiting for $process_name to start.  waited $timeout seconds"
```

Comment:
I would return the numbers of seconds waited as `elapsed`. Just like how other modules do it.